src/jpeg/entropy.rs

```diff
@@ -1,24 +1,26 @@
+#![feature(const_fn)]
+
 /// Given an array containing the number of codes of each code length,
 /// this function generates the huffman codes lengths and their respective
 /// code lengths as specified by the JPEG spec.
-fn derive_codes_and_sizes(bits: &[u8]) -> (Vec<u8>, Vec<u16>) {
-    let mut huffsize = vec![0u8; 256];
-    let mut huffcode = vec![0u16; 256];
+const fn derive_codes_and_sizes(bits: &[u8]) -> ([u8; 256], [u16; 256]) {
+    let mut huffsize = [0u8; 256];
+    let mut huffcode = [0u16; 256];
 
     let mut k = 0;
-    let mut j;
 
     // Annex C.2
     // Figure C.1
     // Generate table of individual code lengths
-    for i in 0u8..16 {
-        j = 0;
-
-        while j < bits[usize::from(i)] {
+    let mut i = 0;
+    while i < 16 {
+        let mut j = 0;
+        while j < bits[i as usize] {
```

Comment:
Neither this. Casting with `as` works for more cases than `usize::from`, including some lossy cases. `clippy` also remarks about this.